src/store/data-slice.js

```diff
@@ -0,0 +1,71 @@
+import { createSlice} from '@reduxjs/toolkit';
+
+
+
+const initialState={
+    data:[],
+    pageCount:0
+}
+
+
+
+const dataSlice = createSlice({
+    name: 'data',
+    initialState,
+    reducers: {
+      setData(state,action){
+        state.data=action.payload.fetchedData
+        state.pageCount=action.payload.pageCount
+      },
+       clearData(state,action){
+        Object.assign(state,initialState);
+        window.location.reload(false);
+       }
+      
+    },
+  });
+  
+  
+
+
+export const fetchData=(url)=>{
+
+    return async (dispatch)=>{
+
+     const  getDataRequest=async()=>{
+      const response = await fetch(url);
+      const data = await response.json();
```

Comment:
Do we need `data` variable?

Reply:
this is our data-slice. we get the data we want from api. also using data for this specific part is common. somehow a convention.